simphony/testing/abc_check_engine.py

```diff
@@ -35,46 +45,48 @@ def create_dataset_items(self):
         """
 
     @abc.abstractmethod
-    def check_emtpy_dataset(self, ds):
-        """ Check if a dataset is empty
-        """
-
-    @abc.abstractmethod
     def check_instance_of_dataset(self, ds):
         """ Check if a dataset is instance of a class
         """
 
-    @abc.abstractmethod
-    def check_is_usable_dataset(self, ds, items):
-        """ Check if a dataset is usable
-        """
+    operation_mapping = {
+        'compare datasets': 'none',
+        'add item': 'none'}
+
+    def compare_operation(self, *args, **kwrds):
+        method = getattr(
+            sys.modules[__name__],
+            self.operation_mapping['compare datasets'])
+        return method(*args, **kwrds)
+
+    def add_operation(self, container, *args, **kwrds):
+        method = getattr(
+            sys.modules[__name__],
+            self.operation_mapping['add item'])
+        return method(*args, **kwrds)
 
     def test_get_missing_dataset(self):
-        filename = os.path.join(self.temp_dir, 'test.cuds')
-        with closing(H5CUDS.open(filename)) as handle:
+        with closing(self.container_factory('test.cuds')) as handle:
```

Comment:
> This template should be independent of the H5CUDS class.

As mentioned in #182, this needs to be independent of H5CUDS.

I think you need a